src/shapes/disk.cpp

```diff
@@ -208,16 +220,24 @@ class Disk final : public Shape<Float, Spectrum> {
         Float cos_phi = select(neq(r, 0.f), local_x * inv_r, 1.f),
               sin_phi = select(neq(r, 0.f), local_y * inv_r, 0.f);
 
-        si.dp_du      = m_to_world * Vector3f( cos_phi, sin_phi, 0.f);
-        si.dp_dv      = m_to_world * Vector3f(-sin_phi, cos_phi, 0.f);
+        si.dp_du[active] = m_to_world * Vector3f(cos_phi, sin_phi, 0.f);
+        si.dp_dv[active] = m_to_world * Vector3f(-sin_phi, cos_phi, 0.f);
 
-        si.n          = m_frame.n;
-        si.sh_frame.n = m_frame.n;
-        si.uv         = Point2f(r, v);
-        si.p          = ray_(si.t);
-        si.time       = ray_.time;
+        si.n[active]            = m_frame.n;
+        si.sh_frame.n[active]   = m_frame.n;
+        si.uv[active]           = Point2f(r, v);
+        si.p[active]            = ray_(si.t);
+        masked(si.time, active) = ray_.time;
+
+        return si;
+    }
 
-        si_out[active] = si;
+    std::pair<Point3f, Normal3f> differentiable_position(const SurfaceInteraction3f &si,
+                                                         Mask /*active*/) const override {
+        auto [c, s] = sincos(si.uv.y() * math::TwoPi<Float>);
```

Comment:
Should it be `[s, c]` instead of `[c, s]`?
```cpp
auto [c, s] = sincos(si.uv.y() * math::TwoPi<Float>); // wrong?
auto [s, c] = sincos(si.uv.y() * math::TwoPi<Float>); // correct?
```